code/game/objects/mail.dm

```diff
@@ -0,0 +1,233 @@
+	/obj/item/toy/figure/crew/chaplain,
+	/obj/item/toy/figure/crew/janitor,
+	/obj/item/toy/figure/crew/librarian,
+	/obj/item/toy/figure/crew/mime)
+	job_list = list("Head of Personnel", "Bartender", "Chef", "Cook", "Culinary Artist", "Butcher", "Botanist", "Hydroponicist", "Botanical Researcher", "Clown", "Mime", "Janitor", "Custodial Technician", "Librarian", "Journalist", "Barber")
+
+/obj/item/envelope/command
+	icon_state = "mail_com"
+	possible_contents = list(/obj/item/flash,
+	/obj/item/storage/fancy/cigarettes/cigpack_robustgold,
+	/obj/item/poster/random_official,
+	/obj/item/book/manual/wiki/sop_command,
+	/obj/item/reagent_containers/food/pill/patch/synthflesh,
+	/obj/item/paper_bin/nanotrasen,
+	/obj/item/reagent_containers/food/snacks/spesslaw,
+	/obj/item/clothing/head/collectable/petehat,
+	/obj/item/toy/figure/crew/captain,
+	/obj/item/toy/figure/crew/lawyer,
+	/obj/item/toy/figure/crew/dsquad)
+	job_list = list("Captain", "Magistrate", "Nanotrasen Representative", "Blueshield", "Internal Affairs Agent", "Human Resources Agent")
+
+/obj/item/envelope/misc
+	possible_contents = list(/obj/item/clothing/under/misc/assistantformal,
+	/obj/item/clothing/under/syndicate/tacticool,
+	/obj/item/clothing/shoes/ducky,
+	/obj/item/toy/plushie/orange_fox/grump,
+	/obj/item/multitool,
+	/obj/item/instrument/piano_synth,
+	/obj/item/toy/crayon/spraycan,
+	/obj/item/clothing/head/cakehat,
+	/obj/item/toy/figure/crew/assistant,
+	/obj/item/toy/figure/owl,
+	/obj/item/toy/figure/griffin)
+	job_list = list("Assistant", "Explorer")
+
+/obj/item/mail_scanner
+	name = "mail scanner"
+	desc = "Confirms deliveries of mail with clients."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "mail_scanner"
+	flags = CONDUCT
+	slot_flags = SLOT_BELT
+	w_class = WEIGHT_CLASS_SMALL
+	origin_tech = "magnets=1"
+	var/obj/item/envelope/saved
+
+/obj/item/mail_scanner/attack()
+	return
+
+/obj/item/mail_scanner/afterattack(atom/A, mob/user)
+	if(istype(A, /obj/item/envelope))
+		var/obj/item/envelope/envelope = A
+		if(envelope.has_been_scanned)
+			to_chat(user, "<span class='warning'>This letter has already been logged to the active database!</span>")
+			playsound(loc, 'sound/machines/deniedbeep.ogg', 50, TRUE)
+			return
+		to_chat(user, "<span class='notice'>You add [envelope] to the active database.</span>")
+		playsound(loc, 'sound/mail/mailscanned.ogg', 50, TRUE)
+		saved = A
+		return
+	if(isliving(A))
+		var/mob/living/M = A
+		if(!saved)
+			to_chat(user, "<span class='warning'>You have not logged mail to the mail scanner!</span>")
+			playsound(loc, 'sound/mail/maildenied.ogg', 50, TRUE)
+			return
+
+		if(M.stat == DEAD)
+			to_chat(user, "<span class='warning'>You can't deliver mail to a corpse!</span>")
+			playsound(loc, 'sound/mail/maildenied.ogg', 50, TRUE)
+			return
+
+		if(M != saved.recipient)
+			to_chat(user, "<span class='warning'>The scanner will not accept confirmation of orders from non clients!</span>")
```

Comment:
This message isn't particularly clear -- can you change it to make it clearer what it's trying to get across?

Reply:
I think the message just has some confusing wording. 

> `The scanner will not accept confirmation of orders from non clients!` 
> orders from non-clients

Is the recipient a client? A client of what? Did they order something, or was it the mysterious sender?

You could probably just make it something like 'this mail isn't addressed to this person!' or 'identity verification failed: target is not an authorized recipient of this package' or something.

Reply:
Changed in upcoming commit.